src/Features/CSharp/Portable/Completion/KeywordRecommenders/ReadOnlyKeywordRecommender.cs

```diff
@@ -41,5 +42,11 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
         private static bool IsRefReadOnlyContext(CSharpSyntaxContext context)
             => context.TargetToken.IsKind(SyntaxKind.RefKeyword) &&
                context.TargetToken.Parent.IsKind(SyntaxKind.RefType);
+
+        private static bool IsValidContextForType(CSharpSyntaxContext context, CancellationToken cancellationToken)
+        {
+            return context.IsTypeDeclarationContext(validModifiers: SyntaxKindSet.AllTypeModifiers,
+                validTypeDeclarations: SyntaxKindSet.ClassStructTypeDeclarations, canBePartial: false, cancellationToken);
```

Comment:
can't have a partial readonly struct?

can you have a readonly class?  should this just be valid on StructTypeDecls?

Reply:
Good catch. Fixed the partial case.

`validTypeDeclarations` is correct though. It checks the containing type declaration. A readonly struct can be declared inside a class or a struct (there are tests for that).